etna/transforms/timestamp/event.py

```diff
@@ -0,0 +1,155 @@
+    ):
+        """
+        Init EventTransform.
+
+        Parameters
+        ----------
+        in_column:
+            binary column with event indicator.
+        out_column:
+            base for creating out columns names.
+        n_pre:
+            number of days before the event to react.
+        n_post:
+            number of days after the event to react.
+        mode: {'binary', 'distance'}, default='binary'
+            Specify mode of marking events:
+
+            - `'binary'`: whether there will be or were events regarding current date in binary type;
+            - `'distance'`: distance to the previous and future events regarding current date;
+
+        Raises
+        ------
+        TypeError:
+            Type of `n_pre` or `n_post` is different from `int`.
+        TypeError:
+            Value of `mode` is not in ['binary', 'distance'].
+        """
+        if not isinstance(n_pre, int) or not isinstance(n_post, int):
+            raise TypeError('`n_pre` and `n_post` must have type `int`')
+        if mode not in ['binary', 'distance']:
+            raise TypeError(f'{type(self).__name__} supports only modes in [\'binary\', \'distance\'], got {mode}.')
+
+        super().__init__(required_features=[in_column])
+        self.in_column = in_column
+        self.out_column = out_column
+        self.n_pre = n_pre
+        self.n_post = n_post
+        self.mode = mode
+        self.in_column_regressor: Optional[bool] = None
+
+    def fit(self, ts: TSDataset) -> "EventTransform":
+        """Fit the transform."""
+        self.in_column_regressor = self.in_column in ts.regressors
+        super().fit(ts)
+        return self
+
+    def _fit(self, df: pd.DataFrame):
+        """Fit method does nothing and is kept for compatibility.
+
+        Parameters
+        ----------
+        df:
+            dataframe with data.
+        """
+        pass
+
+    def _transform(self, df: pd.DataFrame) -> pd.DataFrame:
+        """Mark days before and after event.
+
+        Parameters
+        ----------
+        df:
+            dataframe with data to transform.
+
+        Returns
+        -------
+        :
+            transformed dataframe
+
+        """
+        indexes = df.copy()
+        indexes[:] = np.repeat((np.arange(len(indexes)) + 1).reshape(-1, 1), len(indexes.columns), axis=1)
+
+        prev = df.copy()
+        prev.mask(prev != 1, None, inplace=True)
+        prev = prev * indexes
+        prev = prev.bfill().fillna(indexes)
+        prev = prev - indexes
+        if self.mode == 'binary':
+            prev.mask((prev >= 1) & (prev <= self.n_pre), 1, inplace=True)
+            prev.mask(prev > self.n_pre, 0, inplace=True)
+        else:
+            prev.mask(prev > self.n_pre, 0, inplace=True)
+            prev.mask((prev >= 1) & (prev <= self.n_pre), 1 / prev, inplace=True)
+        prev.rename(columns={self.in_column: f'{self.out_column}_prev'}, inplace=True, level="feature")
+
+        post = df.copy()
```

Comment:
Try to create the separate method to cover this logic, it is mostly the same for prev and post -- try to avoid code duplication